content/blog/2019-10-3-tdd-gimmick/index.md

```diff
@@ -0,0 +1,283 @@
+---
+title: 'Is TDD just a gimmick?'
+date: '2019-03-24T12:51:00.314Z'
+---
+
+# Is TDD A Gimmick?
+
+[[toc]]
+
+## What's the big deal?
+
+Something that has radically improved the quality and maintainability of the software I've written over the past year has been `Test-Driven Development` (TDD for short). My claim is that writing the test **before** the implementation is not just a gimmick:
+
+> Unit tests produced by TDD guarantee code-coverage in a way that tests written post-implementation cannot.
+
+As developers I understand that there's a healthy amount of skepticism we all have so let's go through some of the common responses I get when people hear that claim.
+
+## Maybe it does, but why should I care about code-coverage?
+
+Just to be clear, 100% code-coverage means that if a function is performing `N` operations, each of them are covered by a unit-test. Put another way, if someone were to delete[^del] a line of code atleast one unit-test _should_ fail.
+
+If we have coverage we can have confidence, and with that confidence you get some effects:
+
+- effort to comprehend a piece of code will always grow more slowly than the complexity of that code, because:
+  - there's always going to be atleast one example of every potential usage, and
```

Comment:
typo: atleast --> at<blank>least

Reply:
is it just me or do those two look the same? 🙈 

I think you mean `at-least` or `at least` instead.

Reply:
lol. I meant at least. I typed `at<blank>least` and Github decided to clean up my input.